src/prefect/workers/base.py

```diff
@@ -254,6 +254,8 @@ def _related_resources(self) -> List[RelatedResource]:
         related = []
 
         for kind, obj in self._related_objects.items():
+            if not obj:
```

Comment:
👍 Sure

Reply:
For additional context, this is more of an issue when working with user parameters because some types do weird things when cast to a bool. For example, `if <dataframe>` will throw a warning saying you shouldn't check for truth that way.

Reply:
Thanks for that context! I'll try to prefect `is not None` in general.